src/auth/passport.js

```diff
@@ -79,5 +82,43 @@ const authenticate = (strategy, req, res, next) =>
     })(req, res, next);
   });
 
-export { authenticate };
+passport.use(
+  new GoogleStrategy(
+    {
+      clientID: config.social.google.clientId,
+      clientSecret: config.social.google.clientSecret,
+      callbackURL: 'http://localhost:8080/auth/google/callback',
+    },
+    (accessToken, refreshToken, profile, done) => {
+      const email = profile.emails[0].value;
+      User.findOne({ email })
+        .then(async user => {
+          if (user) {
+            return user;
+          }
+          const firstName = { unknown: profile.name && profile.name.givenName };
+          const lastName = { unknown: profile.name && profile.name.familyName };
+          const imageUrl = cutSizing(profile.photos && profile.photos[0].value);
```

Comment:
is't any other query params? we can made more generic method:

```js
const formatUrl = url => url && \\ cut all query params
```

and, please, place it not in `validation`, cause it not validator, but helper / formatter